app/[lang]/page.tsx

```diff
@@ -11,16 +11,24 @@ import {
 } from '@/app/[lang]/dictionaries';
 import { useRouter } from 'next/navigation';
 import type { AppRouterInstance } from 'next/dist/shared/lib/app-router-context.shared-runtime';
+import { useSession } from 'next-auth/react';
 
 const Home = ({ params: { lang } }: { params: { lang: string } }) => {
   const dict: Dict = getDictionary(isLang(lang) ? lang : defaultLocale);
   const router: AppRouterInstance = useRouter();
+  const { data: session, status, update } = useSession();
```

Comment:
Ideally this should be serverside, but I do see that you would then need to rewrite the language cookie logic to be serverside too, so it's fine to leave this for now